source/Halibut/Transport/SecureListener.cs

```diff
@@ -230,14 +226,7 @@ async Task ExecuteRequest(TcpClient client)
                         });
 
                         tcpClientManager.AddActiveClient(thumbprint, client);
-                        // Delegate the open stream to the protocol handler - we no longer own the stream lifetime
                         await ExchangeMessages(ssl);
-
-                        if (verifyClientThumbprint(thumbprint))
-                        {
-                            // Mark the stream as delegated once everything has succeeded
-                            keepStreamOpen = true;
```

Comment:
This was the bug... When a conversation ended correctly, we were leaving the stream/client open and never closing it.